sykepenger-mediators/src/test/kotlin/no/nav/helse/spleis/e2e/PostgresDatabase.kt

```diff
@@ -1,113 +1,43 @@
-        Runtime.getRuntime().addShutdownHook(Thread(this::stop))
-    }
-
-    private fun createSchema(dataSource: DataSource) {
-        Flyway.configure().dataSource(dataSource).load().migrate()
-        using(sessionOf(dataSource)) { it.run(queryOf(truncateTablesSql).asExecute) }
-    }
-
-    private fun resetSchema() {
-        using(sessionOf(connection())) { it.run(queryOf("SELECT truncate_tables();").asExecute) }
-    }
-
-    private fun stopDatabase() {
-        embeddedPostgres!!.close()
-        embeddedPostgres = null
-    }
-
-    private fun createHikariConfig(jdbcUrl: String) =
-        HikariConfig().apply {
-            this.jdbcUrl = jdbcUrl
-            maximumPoolSize = 3
-            minimumIdle = 1
-            idleTimeout = 10001
-            connectionTimeout = 1000
-            maxLifetime = 30001
-        }
-
-    private interface DBState {
-        fun connection(db: PostgresDatabase): DataSource {
-            throw IllegalStateException("Cannot create connection in state ${this::class.simpleName}")
+object PostgresContainer {
+    val instance by lazy {
+        PostgreSQLContainer<Nothing>("postgres:12.1").apply {
+            start()
         }
-        fun start(db: PostgresDatabase) {}
-        fun stop(db: PostgresDatabase) {}
-        fun reset(db: PostgresDatabase) {}
     }
+}
 
-    private object NotStarted : DBState {
-        override fun start(db: PostgresDatabase) {
-            state = Started
-            db.startDatbase()
+object SpleisDataSource {
+    private val instance: HikariDataSource by lazy {
+        HikariDataSource().apply {
+            username = PostgresContainer.instance.username
+            password = PostgresContainer.instance.password
+            jdbcUrl = PostgresContainer.instance.jdbcUrl
+            connectionTimeout = 1000L
         }
     }
 
-    private object Started : DBState {
-        override fun stop(db: PostgresDatabase) {
-            db.state = NotStarted
-            db.stopDatabase()
-        }
-
-        override fun connection(db: PostgresDatabase): DataSource {
-            return db.dataSource!!
-        }
-
-        override fun reset(db: PostgresDatabase) {
-            db.resetSchema()
-        }
-
-    }
+    val migratedDb = instance.also { migrate(it) }
+}
 
-    @Language("PostgreSQL")
-    private val truncateTablesSql = """
-CREATE OR REPLACE FUNCTION truncate_tables() RETURNS void AS ${'$'}${'$'}
-DECLARE
-    statements CURSOR FOR
-        SELECT tablename FROM pg_tables
-        WHERE schemaname = 'public' AND tablename NOT LIKE 'flyway%';
-BEGIN
-    FOR stmt IN statements LOOP
-        EXECUTE 'TRUNCATE TABLE ' || quote_ident(stmt.tablename) || ' CASCADE;';
-    END LOOP;
-END;
-${'$'}${'$'} LANGUAGE plpgsql;
-"""
+private val tabeller = listOf("person", "melding", "unike_person")
+fun resetDatabase() {
+    using(sessionOf(migratedDb)) { session -> tabeller.forEach { table -> session.run(queryOf("truncate $table cascade").asExecute) } }
```

Comment:
Jeg mener vi fant noe ytelsesgull i den håndsmidde trunkeringen, den ble innført (i spesialist) her: https://github.com/navikt/helse-spesialist/commit/39d6e1a96cdd1856b8c05cbea632e139921cc189

Men det er kanskje mindre viktig for mediators, vi kjører jo ikke de testene så ofte lokalt og det er ikke like mange tester som spesialist hadde og/eller har heller.